### PR TITLE
Add method isVersioningEnabled()

### PR DESCRIPTION
**Pull request by sinhaashish**
Added the `isVersioningEnabled` which return true if bucket is version enabled.

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -3648,9 +3649,8 @@ public void enableVersioning(EnableVersioningArgs args)
 
     Map<String, String> queryParamMap = new HashMap<>();
     queryParamMap.put("versioning", "");
-    String config =
-        "<VersioningConfiguration xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">"
-            + "<Status>Enabled</Status></VersioningConfiguration>";
+    VersioningConfiguration config = new VersioningConfiguration("Enabled");
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java:3652]: Its better to pass a `boolean` rather a string.

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -3712,13 +3712,58 @@ public void disableVersioning(DisableVersioningArgs args)
...
+    checkArgs(args);
+
+    Map<String, String> queryParamMap = new HashMap<>();
+    queryParamMap.put("versioning", "");
+    try (Response response = executeGet(args.bucket(), null, null, queryParamMap)) {
+      VersioningConfiguration result =
+          Xml.unmarshal(VersioningConfiguration.class, response.body().charStream());
+      return result.status();
+    } catch (ErrorResponseException e) {
+      if (e.errorResponse().errorCode() != ErrorCode.ILLEGAL_VERSIONING_CONFIGURATION_EXCEPTION) {
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java:3760]: When do we get this error?

File: api/src/main/java/io/minio/messages/VersioningConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+package io.minio.messages;
+
+import org.simpleframework.xml.Element;
+import org.simpleframework.xml.Namespace;
+import org.simpleframework.xml.Root;
+
+/**
+ * Object representation of request XML of <a
+ * href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketVersioning.html">PutBucketVersioning
```
Comment (balamurugana) [inline api/src/main/java/io/minio/messages/VersioningConfiguration.java:25]: I don't think the link going to work. Have you checked it?

File: api/src/main/java/io/minio/messages/VersioningConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+@Namespace(reference = "http://s3.amazonaws.com/doc/2006-03-01/")
+public class VersioningConfiguration {
+  @Element(name = "Status", required = false)
+  private String status;
+
+  public VersioningConfiguration() {}
+
+  /** Constructs a new VersioningConfiguration object with given status. */
+  public VersioningConfiguration(String status) {
+    this.status = status;
```
Comment (balamurugana) [inline api/src/main/java/io/minio/messages/VersioningConfiguration.java:40]: Accept `boolean` and construct status string accordingly

File: api/src/main/java/io/minio/messages/VersioningConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  /** Indicates whether the bucket is version enabled or not. */
+  public boolean status() {
+    return status != null && status.equals("Enabled");
```
Comment (balamurugana) [inline api/src/main/java/io/minio/messages/VersioningConfiguration.java:45]: `return "Enabled".equals(status)` would work

File: docs/API.md [hunk 6]
```diff
@@ -23,23 +23,24 @@ MinioClient s3Client = new MinioClient("https://s3.amazonaws.com",
...
-| [`listIncompleteUploads`](#listIncompleteUploads)             | [`presignedPutObject`](#presignedPutObject)             |
-| [`makeBucket`](#makeBucket)                                   | [`putObject`](#putObject)                               |
-| [`removeAllBucketNotification`](#removeAllBucketNotification) | [`removeObject`](#removeObject)                         |
-| [`removeBucket`](#removeBucket)                               | [`removeObjects`](#removeObjects)                       |
-| [`removeIncompleteUpload`](#removeIncompleteUpload)           | [`selectObjectContent`](#selectObjectContent)           |
-| [`setBucketEncryption`](#setBucketEncryption)                 | [`setObjectRetention`](#setObjectRetention)             |
-| [`setBucketLifeCycle`](#setBucketLifeCycle)                   | [`setObjectTags`](#setObjectTags)                       |
-| [`setBucketNotification`](#setBucketNotification)             | [`statObject`](#statObject)                             |
+| [`isVersioningEnabled`](#isVersioningEnabled)                 | [`getObject`](#getObject)                               |
```
Comment (balamurugana) [inline docs/API.md:26]: Keep the list in columns sorted.

File: api/src/main/java/io/minio/messages/VersioningConfiguration.java (hunk 3 above)
Comment (sinhaashish) [reply api/src/main/java/io/minio/messages/VersioningConfiguration.java:25]: It works.
Screenshot from 2020-05-29 11-04-11

File: api/src/main/java/io/minio/MinioClient.java (hunk 2 above)
Comment (sinhaashish) [reply api/src/main/java/io/minio/MinioClient.java]: Removed

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (sinhaashish) [reply api/src/main/java/io/minio/MinioClient.java]: done

File: api/src/main/java/io/minio/messages/VersioningConfiguration.java (hunk 3 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/messages/VersioningConfiguration.java:25]: The illusion I saw was, `href` string was broken into next line.

**Comment by sinhaashish** [comment]
> Please add an example

Added.

File: functional/FunctionalTest.java [hunk 7]
```diff
@@ -4880,6 +4860,7 @@ public static void runQuickTests() throws Exception {
   public static boolean downloadMinio() throws IOException {
     String url = "https://dl.min.io/server/minio/release/";
     if (OS.contains("linux")) {
+
```
Comment (anjalshireesh) [inline functional/FunctionalTest.java:4863]: This empty line can be removed

File: docs/API.md [hunk 8]
```diff
@@ -31,20 +31,22 @@ MinioClient s3Client = new MinioClient("https://s3.amazonaws.com",
...
+| [`removeBucket`](#removeBucket)                               | [`selectObjectContent`](#selectObjectContent)           |
+| [`removeIncompleteUpload`](#removeIncompleteUpload)           | [`setObjectRetention`](#setObjectRetention)             |
+| [`setBucketEncryption`](#setBucketEncryption)                 | [`setObjectTags`](#setObjectTags)                       |
+| [`setBucketLifeCycle`](#setBucketLifeCycle)                   | [`statObject`](#statObject)                             |
+| [`setBucketNotification`](#setBucketNotification)             |                                                         |
 | [`setBucketPolicy`](#setBucketPolicy)                         |                                                         |
 | [`setBucketTags`](#setBucketTags)                             |                                                         |
 | [`setDefaultRetention`](#setDefaultRetention)                 |                                                         |
 
+
```
Comment (balamurugana) [inline docs/API.md:49]: This new line is not required.

**Pull request merged**